src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -52,8 +54,10 @@ class CoreWorkerDirectTaskSubmitter {
   /// Schedule more work onto an idle worker or return it back to the raylet if
   /// no more tasks are queued for submission. If an error was encountered
   /// processing the worker, we don't attempt to re-use the worker.
-  void OnWorkerIdle(const rpc::WorkerAddress &addr, const SchedulingKey &task_queue_key,
-                    bool was_error) EXCLUSIVE_LOCKS_REQUIRED(mu_);
+  void OnWorkerIdle(
+      const rpc::WorkerAddress &addr, const SchedulingKey &task_queue_key, bool was_error,
+      const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> resource_mapping)
```

Comment:
Suggest adding a unit test to make sure the resource_mapping stays associated with the right worker?